clients/src/test/java/org/apache/kafka/clients/MockClient.java

```diff
@@ -245,10 +246,18 @@ public void send(ClientRequest request, long now) {
                 unsupportedVersionException = new UnsupportedVersionException(
                         "Api " + request.apiKey() + " with version " + version);
             } else {
-                AbstractRequest abstractRequest = request.requestBuilder().build(version);
-                if (!futureResp.requestMatcher.matches(abstractRequest))
-                    throw new IllegalStateException("Request matcher did not match next-in-line request "
-                            + abstractRequest + " with prepared response " + futureResp.responseBody);
+                try {
+                    AbstractRequest abstractRequest = request.requestBuilder().build(version);
+                    if (!futureResp.requestMatcher.matches(abstractRequest))
+                        throw new IllegalStateException("Request matcher did not match next-in-line request "
+                                + abstractRequest + " with prepared response " + futureResp.responseBody);
+                } catch (NoBatchedFindCoordinatorsException uble) {
```

Comment:
Yes I looked into it but I think there's 1 test in TransactionManagerTest that expects `UnsupportedVersionException` to be thrown.
MockClient works in slightly different ways than the real client and it would be good to address this but I'd rather defer to a follow up PR.

Reply:
Understood. Should we file a Jira about this?

Reply:
Sure, I raised https://issues.apache.org/jira/browse/KAFKA-13000